src/Adapter/Http/PodStatusProvider.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Kubernetes\Client\Adapter\Http;
+
+use Kubernetes\Client\Exception\PodNotFound;
+use Kubernetes\Client\Model\ContainerStatus;
+use Kubernetes\Client\Model\Pod;
+use Kubernetes\Client\Model\PodAwareStatusProvider;
+use Kubernetes\Client\Model\PodStatus;
+use Kubernetes\Client\Repository\PodRepository;
+
+/**
+ * Provide pod status using a pod repository to fetch actual data.
+ */
+class PodStatusProvider implements PodAwareStatusProvider
+{
+    /**
+     * @var Pod
+     */
+    private $pod;
+
+    /**
+     * @var PodRepository
+     */
+    private $podRepository;
+
+    public function __construct(PodRepository $podRepository)
+    {
+        $this->podRepository = $podRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPod(Pod $pod)
```

Comment:
It looks like to me this should be a constructor's argument

Reply:
That was my initial thought, but the provider class could not be reused when the `$pod` instance changes.
I would rather pass the `$pod` argument to the `isTerminated()` and `isPending()` methods, to avoid creating new instances.